### ISSUE TITLE
AnimationPlayer "Loop" property working sporadically

### ISSUE THREAD

**Issue by Ceryni** [godotengine/godot#25245]
**Godot version:**
3.1 beta 2

**OS/device including version:**
Windows 10

**Issue description:**
AnimationPlayer not respecting the "loop" setting. Plays animation only once. I've tried setting the loop property via code as well to no avail.

**Steps to reproduce:**
Setup an animation that is supposed to loop. Seems to work sporadically. Sometimes it will loop correctly, but most times it will not.

**Minimal reproduction project:**
I noticed it while going through this tutorial, so here's the project I've got so far.
[Zeldalike-Tutorial.zip](https://github.com/godotengine/godot/files/2785846/Zeldalike-Tutorial.zip)

**Comment by Ceryni** [comment]
When I move the player sprite around, the animations don't seem to be looping. They play once and do not continue playing. I recreated the project on the stable 3.0.6 release and it worked as expected.

Are the animations playing and looping correctly for you?

**Comment by avencherus** [comment]
I can confirm this in a recent master build from yesterday.  Most likely a bug, can't find anything wrong with the setup, even tried changing the processing to physics.

The animation in game reports that it is running and gives times.  However, the properties on the Sprite indicate it changes once and gets trapped.  (Seems to be on any property)

![image](https://user-images.githubusercontent.com/13004169/52101836-1534d300-25e6-11e9-9a8a-6dc7645d8262.png)

This is even visible my AnimationPlayer.  I can only get it to play again by doing a restart on the animation, but while it is loop it doesn't animate.

![ezgif com-video-to-gif 2](https://user-images.githubusercontent.com/13004169/52101846-21b92b80-25e6-11e9-90d0-7d78310e9b6c.gif)

Maybe some issue with the loop?

This poor 3.1 AnimationPlayer.  I don't think it liked the rewrites.  X)

@akien-mga It probably should be aimed for 3.1, it may mean that no looping animations for any users that run into this until 3.2.

**Comment by avencherus** [comment]
@Ceryni Does setting the AnimationPlayer playback speed of 3.0 fix the problem on your end?

I think the problem might be to do with the speed.

**Comment by Ceryni** [comment]
@avencherus Confirmed, switching it to 3.0 fixes it and going back to 4.0 makes it break again.

**Comment by Ceryni** [comment]
Tried 2,3,4,5,6, Only multiplier that had issues was 4.

**Comment by akien-mga** [comment]
Can't reproduce in 88a36e20 on Linux, so assuming fixed.

Please comment if you can still reproduce the issue.

**Comment by avencherus** [comment]
Was not fixed.  It seems to occur less easily, but still happens for me in Windows 10 64-bit as of a32b26d.

![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/13004169/53392349-c8080f00-39a1-11e9-9660-8bc7ec68fb3b.gif)
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/13004169/53392354-c9393c00-39a1-11e9-9981-b39f6f2dc69e.gif)

**Comment by akien-mga** [comment]
Weird. Might be Windows specific?

**Comment by avencherus** [comment]
No, took a look just now.  On my laptop using Ubuntu 18.04.1 LTS, I also have that odd stalling of the animation when changing directions, and the AnimationPlayer can hiccup as well, though less often.

**Comment by reduz** [comment]
I can't reproduce this at all, will have to test on Windows..

**Comment by reduz** [comment]
Aha, I can reproduce it on Windows

**Comment by takhimi** [comment]
@reduz 
Using Beta10 and Windows10 64bit. 
For me the "Animation Looping" when ticked in the animation editor and click the play animation button, it will loops. But if I click on Play Project or Play scene, it wont loops. If I close the project and reopen it back the "Animation Looping" is not ticked. The "Autoplay on load" is working just fine.

**Comment by avencherus** [comment]
You shouldn't see the changes in beta10, this commit happened about 15 hours after beta10's.

**Comment by jesperkondrup** [comment]
@avencherus Yep it definitely looks like the same issue but in my case I found a work around by setting the step value to 0.1.

**Comment by avencherus** [comment]
@jesperkondrup Odd, I'll have to keep an eye out for this.  Was it editor side only?  Since the step is for keyframe snapping.

**Comment by jesperkondrup** [comment]
@avencherus I just took a look and it seems that this is only happening in the editor.

**Comment by jesperkondrup** [comment]
I am not comlpetely sure if this is related but recently we had this issue in some of our animations (in 3D) that some animations that triggered other animations would not play the triggered aniamation - BUT this was only happening the first time the animation was played. All the subsequently times the animation was played it triggered the subanimations just fine. 

This time the step value fix of using 0.1 instead of 0.01 did not work but instead we made a call method track and added the animation that should be triggered through there. This seems to do the trick and now it works every time.

So what is the difference between calling the animation where you can see the track in the timeline and the method track? It would seem the Method track is more reliable when you want to trigger an animation?

Also my colleague on linux cannot reproduce the issue (I am working on Windows 10).

**Comment by avinashbn2** [comment]
Facing the same issue on ubuntu with godot  3.2.2
Looping doesn't work when i keep holding the input but works when i keep pressing & releasing the key

**Comment by starry-abyss** [comment]
Maybe worth reopening then?

**Comment by drentsoft** [comment]
Having the same issue in 3.2.3.stable. Win 10 Pro x64. Looped animations play fine in the editor but in the game they play once then stall. If I set get_animation(name).loop = true in code then it honours it. I've attached a minimal reproduction project.
[AnimationLoopIssue.zip](https://github.com/godotengine/godot/files/5377480/AnimationLoopIssue.zip)

**Comment by avencherus** [comment]
> Having the same issue in 3.2.3.stable. Win 10 Pro x64. Looped animations play fine in the editor but in the game they play once then stall. If I set get_animation(name).loop = true in code then it honours it. I've attached a minimal reproduction project.
> [AnimationLoopIssue.zip](https://github.com/godotengine/godot/files/5377480/AnimationLoopIssue.zip)

This seems like a different issue, one related to .glb imports.

It looks like the problem in your sample project is that changing loop properties in the editor aren't acknowledged when using an .glb import.  From what I can tell no edits or additions to keyframes appear to be stored either.

The expectation might be to change these things from the source and reimport it.  I couldn't speak to the intended ability of the AnimationPlayer to take the data and modify these files from the editor.  If it's a bug you should open a new issue, but if not, it might be a feature request to allow AnimationPlayer to backwards modify these special import files.

If you make the scene local, these editor changes in the AnimationPlayer become possible, as the data is then captured in the .tscn file from then on.

It does becomes a rather large text file at that point.

![image](https://user-images.githubusercontent.com/13004169/96088424-172fea00-0ece-11eb-8496-2e2d2ea2ab10.png)

**Comment by StalfFromValenwood** [comment]
> 
> This seems like a different issue, one related to .glb imports.
> 
Just to be sure, I tested it with .gltf but the issue remains the same.
However, the -loop sufix seems to work in any case.

**Comment by FilipLundby** [comment]
Adding -loop suffix in Blender to animation name and NLA-track worked for me too. Godot 3.5 beta 2.
